crates/wasi/src/preview2/table.rs

```diff
@@ -67,13 +90,15 @@ impl TableEntry {
 impl Table {
     /// Create an empty table
     pub fn new() -> Self {
+        // TODO: what's a good default here?
+        Table::with_capacity(100)
```

Comment:
I don't have an intuition for what a good magic number for capacity is. @alexcrichton any guesses?

Reply:
Could `with_capacity` be removed and this start as an empty vec? I'm not sure if there's much to gain from `with_capacity` really other than letting default heuristics take over.

Reply:
My thought here was that it would be good to allow a bit of control so that embedders could make that decision depending on their workloads. Perhaps that's over-estimating the cost of reallocating the vector though.

Reply:
Ok keeping `with_capacity` makes sense to me, but I think that `new` should create with the default 0 capacity of `Vec` and then `with_capacity` sets an initial capacity.